gnd/src/main/java/com/google/android/gnd/persistence/sync/LocalMutationSyncWorker.java

```diff
@@ -112,17 +112,17 @@ private Completable processMutations(ImmutableList<Mutation> mutations, String u
   private Completable processMutations(ImmutableList<Mutation> mutations, User user) {
     return remoteDataStore
         .applyMutations(mutations, user)
-        .andThen(removeLocalObservationsIfMutationTypeIsDelete(mutations))
+        .andThen(deleteObservationIfRemovedRemotely(mutations))
```

Comment:
Why don't we just delete the observation as soon as the remote observation is deleted inside `applyMutations`? Sorry if I'm missing something!

Reply:
https://github.com/google/ground-android/blob/1a1246f0a55404c0fe3ed98ea83c1b35dfc594e4/gnd/src/main/java/com/google/android/gnd/persistence/remote/firestore/FirestoreDataStore.java#L112

IIUC, deleting the observation just after adding batch request would not ensure that the remote observation was indeed removed. So, to ensure that we only do it after successful task completion, I chose to do it in after the success call.
Does that seem right to you?

Reply:
Ah, good point; sorry I forgot we were deleting them as part of the batch. I was concerned about possible race conditions, but we're using the same set of mutations in both steps, so I think we're ok here. LGTM!